### ISSUE TITLE
Prevent line curvature overshooting

### ISSUE THREAD

**Issue by stx** [imaNNeoFighT/fl_chart#25]
Thanks for this plugin. It's really fantastic and you're really killing it!

I love the curved lines, but they cause data confusion due to "overshooting" their data point. Changing curvature smoothness doesn't seem to fix this specific issue.

Flat:

![image](https://user-images.githubusercontent.com/1284134/59556309-16ebab80-8f75-11e9-9525-927e770f2019.png)

Curved:

![image](https://user-images.githubusercontent.com/1284134/59556310-20751380-8f75-11e9-8ec7-b98787734a8c.png)

Becomes especially problematic if you see the actual data points:

![image](https://user-images.githubusercontent.com/1284134/59556368-3fc07080-8f76-11e9-9732-c5702bd7cf62.png)

Can we do something like this crummy Photoshop job?

![image](https://user-images.githubusercontent.com/1284134/59556362-20c1de80-8f76-11e9-9cc9-14bc21766f24.png)

Basically, keep curvature, but prevent overshooting?

**Comment by imaNNeoFighT** [comment]
Hi, pleasure is mine. 
Thats a bad issue, 
I will find a solution ASAP
Thanks for report!

**Comment by imaNNeoFighT** [comment]
did you try set `curveSmoothness` less than 0.35?
(the default value is 0.35, you can decrease it)

**Comment by stx** [comment]
Yes, doesn’t have the same effect. I want curves, just not overshooting curves. Thank you!

**Comment by imaNNeoFighT** [comment]
Yes it doesn't have the same effect, but can be used as temporary solution, until I will be free to work on it.

**Comment by imaNNeoFighT** [comment]
fixed on [0.1.2](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.1.2), added `preventCurveOverShooting` on [LineChartBarData](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/line_chart.md#linechartbardata)

`preventCurveOverShooting = false`
<img src="https://user-images.githubusercontent.com/7009300/60774183-cc99bc80-a125-11e9-8343-924abec2ac18.png" width="300">

`preventCurveOverShooting = true`
<img src="https://user-images.githubusercontent.com/7009300/60774179-c73c7200-a125-11e9-8a69-355ae2464138.png" width="300">

**Comment by imaNNeoFighT** [comment]
It would be nice if you share the result on your sample

**Comment by jamesblasco** [comment]
Thanks for this! It's awesome, but while almost all of my graphs works well there are some that with preventCurveOverShooting = true looks like this (see the right side)
![IMG_1828](https://user-images.githubusercontent.com/19904063/71476147-f0d87b00-27e3-11ea-8003-319cbc8f53ef.jpg)

**Comment by SArshpreetAI** [comment]
Hey I am still facing similar issue
flutter: 3.29.2
fl_chart: ^0.70.2
happening only with web.

<img width="980" alt="Image" src="https://github.com/user-attachments/assets/2566076c-3021-41aa-a945-bb7f1f4627cd" />

I have done 
```
isCurved: true,
preventCurveOverShooting: true,
```
